tests/other_modes.rs

```diff
@@ -62,16 +64,17 @@ fn level_triggered() {
         .modify_with_mode(&reader, Event::readable(reader_token), PollMode::Oneshot)
         .unwrap();
 
-    writer.write_all(&data).unwrap();
+    writer.write(&data).unwrap();
     events.clear();
+    // BUG: Somehow, the notification here is delayed?
```

Comment:
obvious place for a race condition, this might also be some context-switching/scheduling stuff...

did you try to look at how the (FreeBSD) kernel handles this internally?

Reply:
how does our test differ from https://cgit.freebsd.org/src/tree/tests/sys/kqueue/libkqueue/read.c?h=stable/13#n175 ?

Reply:
does it make a practical/behavior difference for the timeout if `None` vs `Some(Duration::from_millis(0))` is specified?

Reply:
I would be in favor of using `None` (= no timeout) here, or some really long timeout (e.g. 10 seconds). As this is the only problem remaining, but overall seems to have nothing to do with this PR as-is, I would also be in favor of merging this and creating a new issue for it.